solutions/page_object/pages/product_page.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+from .base_page import BasePage
+from .locators import ProductPageLocators
+
+
+class ProductPage(BasePage):
+
+    def should_be_able_to_add_product_to_cart(self):
+        self.should_be_product_url()
+        self.should_button_add_to_cart_present()
+
+    def should_be_in_cart(self):
+        self.should_be_right_product_name_in_success_message()
+        self.should_cart_price_be_equal_product_price()
+
+    def should_be_product_url(self):
+        current_url = self.browser.current_url
+        assert re.search(
+            r"catalogue/.+/\?promo=newYear", current_url
+        ), f"Expected URL to contain the word 'catalogue' and product name, but got '{current_url}'."
+
+    def should_button_add_to_cart_present(self):
+        button_add_to_cart = self.is_element_present(*ProductPageLocators.ADD_TO_CART)
```

Comment:
Не лучше ли распаковывать сразу в `is_element_present`? Заодно проще будет параметр функции назвать.

```suggestion
        button_add_to_cart = self.is_element_present(ProductPageLocators.ADD_TO_CART)
```

Reply:
сделала